plugins/core-plugin/src/main/java/com/google/cloud/teleport/plugin/YamlDockerfileGenerator.java

```diff
@@ -37,18 +38,27 @@ public class YamlDockerfileGenerator {
   private YamlDockerfileGenerator() {}
 
   public static void generateDockerfile(
-      String basePythonContainerImage, String yamlTemplateName, File targetDirectory)
+      String basePythonContainerImage,
+      String baseJavaContainerImage,
+      String beamVersion,
+      String pythonVersion,
+      String yamlTemplateName,
+      List<String> otherFiles,
+      File targetDirectory)
       throws IOException, TemplateException {
-
     Configuration freemarkerConfig = new Configuration(Configuration.VERSION_2_3_32);
     freemarkerConfig.setDefaultEncoding("UTF-8");
     freemarkerConfig.setTemplateExceptionHandler(TemplateExceptionHandler.RETHROW_HANDLER);
     freemarkerConfig.setLogTemplateExceptions(true);
     freemarkerConfig.setClassForTemplateLoading(PythonDockerfileGenerator.class, "/");
 
     Map<String, Object> parameters = new HashMap<>();
-    parameters.put("baseContainerImage", basePythonContainerImage);
-    parameters.put("yamlTemplateName", yamlTemplateName + ".yaml");
+    parameters.put("basePythonContainerImage", basePythonContainerImage);
+    parameters.put("baseJavaContainerImage", baseJavaContainerImage);
+    parameters.put("beamVersion", beamVersion);
+    parameters.put("pythonVersion", pythonVersion);
+    parameters.put(
+        "otherFiles", String.join("* ", otherFiles) + (otherFiles.size() > 0 ? "*" : ""));
```

Comment:
This is kind of confusing to me - why do we need the * at the end of this?

`COPY main.py` should work just as well as `COPY main.py*`, right? And there could be times you don't want this.

Should we instead just tell consumers that this follows docker syntax and you can append `*` or other special characters - https://docs.docker.com/reference/dockerfile/#copy

Reply:
It was initially a workaround to copy `requirements.txt` only if it exists, but I suppose there is to reason to append asterisks on the other files. I can remove those.